go-controller/pkg/util/kube.go

```diff
@@ -207,7 +207,7 @@ func ServiceTypeHasClusterIP(service *kapi.Service) bool {
 
 // ServiceTypeHasNodePort checks if the service has an associated NodePort or not
 func ServiceTypeHasNodePort(service *kapi.Service) bool {
-	return service.Spec.Type == kapi.ServiceTypeNodePort || service.Spec.Type == kapi.ServiceTypeLoadBalancer
+	return service.Spec.Type == kapi.ServiceTypeNodePort || (service.Spec.Type == kapi.ServiceTypeLoadBalancer && (service.Spec.AllocateLoadBalancerNodePorts == nil || *service.Spec.AllocateLoadBalancerNodePorts))
```

Comment:
I'm getting headaches just by looking at this expression. (having to do horizontal scrolling is annoying).

Is it possible to abstract the new condition you're adding into a function / predicate ? IMO that would make it more readable.

Possibly something like:
```golang
func  doesServiceHaveLoadBalancerNodePortAllocation(service *k8stypes.Service) bool {
    return service.Spec.AllocateLoadBalancerNodePorts == nil || *service.Spec.AllocateLoadBalancerNodePorts
}
```

Then I'd go as far as adding some newlines to the overall condition:
```golang
return service.Spec.Type == kapi.ServiceTypeNodePort || 
    (service.Spec.Type == kapi.ServiceTypeLoadBalancer && doesServiceHaveLoadBalancerNodePortAllocation(service))
```

Reply:
yes that makes total sense, sorry for the complex matching I did here in one go!